src/lib/network_pool/network_pool_base.ml

```diff
@@ -42,14 +59,30 @@ end)
           (Error.to_string_hum e) ;
         Deferred.unit
 
-  let of_resource_pool_and_diffs resource_pool ~logger ~incoming_diffs =
+  let of_resource_pool_and_diffs resource_pool ~logger ~incoming_diffs
+      ~local_diffs ~tf_diff =
     let read_broadcasts, write_broadcasts = Linear_pipe.create () in
     let network_pool =
       {resource_pool; logger; read_broadcasts; write_broadcasts}
     in
-    Linear_pipe.iter incoming_diffs ~f:(fun diff_and_cb ->
-        apply_and_broadcast network_pool diff_and_cb )
-    |> ignore ;
+    (*proiority: Transition frontier diffs > local diffs > incomming diffs*)
+    Strict_pipe.Reader.Merge.iter
+      [ Strict_pipe.Reader.map tf_diff ~f:(fun diff ->
+            `Transition_frontier_extension diff )
+      ; Strict_pipe.Reader.map local_diffs ~f:(fun diff -> `Local diff)
+      ; Strict_pipe.Reader.map incoming_diffs ~f:(fun diff -> `Incoming diff)
+      ]
+      ~f:(fun diff_source ->
+        match diff_source with
+        | `Incoming diff_and_cb ->
+            apply_and_broadcast network_pool diff_and_cb
+        | `Local diff ->
+            (*Should this be coming from resource pool instead?*)
```

Comment:
what do you mean by this comment?

Reply:
oops this was part of my notes (I usually annotate them with my name). But actually, I was wondering if handling local diffs should be specified in the resource pool (similar to `handle_transition_frontier_diff`). Right now, both the pools simply call `apply_and_broadcast` for locally generated diffs so it doesn't matter but if we wanted to do something that is resource-pool specific before applying and broadcasting then it'd make sense to have it as part of resource pool